action.yml

```diff
@@ -181,12 +181,25 @@ runs:
         sed '/^actual: \(.*\)$/!d; s//compiler=\1/' "$tmp" >>"$GITHUB_OUTPUT"
         sed '/^actual: ghc-\(.*\)$/!d; s//compiler-version=\1/' "$tmp" >>"$GITHUB_OUTPUT"
 
+    - id: stack-path
+      name: Set stack-path outputs
+      working-directory: ${{ inputs.working-directory }}
+      shell: bash
+      run: |
+        stack --no-terminal --stack-yaml ${{ inputs.stack-yaml }} \
+          ${{ steps.setup.outputs.resolver-nightly }} \
+          path | while IFS=:\  read -r name value; do
+          printf '%s: %s\n' "$name" "$value"
+          printf '%s=%s\n' "$name" "$value" >>"$GITHUB_OUTPUT"
+        done
+
     - name: Restore dependencies cache
       id: restore-deps
       uses: actions/cache/restore@v3
       with:
         path: |
-          ~/.stack
+          ${{ steps.stack-path.outputs.stack-root }}
+          ${{ steps.stack-path.outputs.programs }}
```

Comment:
Oh, that's convenient that it was already in the `stack-paths` stuff!

Reply:
Yes. It's actually [documented](https://docs.haskellstack.org/en/stable/stack_root/#location).